File: rb/features.rb
# Echofm SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module EchofmFeatures
  def self.make_feature(name)
    case name
    when "base"
      EchofmBaseFeature.new
    when "ratelimit"
      EchofmRatelimitFeature.new
    when "retry"
      EchofmRetryFeature.new
    when "test"
      EchofmTestFeature.new
    when "timeout"
      EchofmTimeoutFeature.new
    else
      EchofmBaseFeature.new
    end
  end
end

File: py/features.py
# Echofm SDK feature factory

from feature.base_feature import EchofmBaseFeature
from feature.test_feature import EchofmTestFeature


def _make_feature(name):
    features = {
        "base": lambda: EchofmBaseFeature(),
        "test": lambda: EchofmTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
